src/nupic/engine/Spec.hpp

```diff
@@ -54,15 +54,16 @@ namespace nupic
     UInt32 count; 
     // TBD. Omit? what is "required"? Is it ok to be zero length?
     bool required;
-    bool regionLevel;
+    bool regionLevel; // As of 6/20/16, optional and default value is true
     bool isDefaultInput;
-    bool requireSplitterMap; // 
+    bool requireSplitterMap; // As of 6/20/16, optional and default value is false
   };
 
   class OutputSpec
   {
   public:
     OutputSpec() {}
+    // 6/20/16 regionLevel is optional and has a default value of true.
```

Comment:
Oops, yeah that documentation is not precise. It isn't optional at the level of the constructor in Spec. They are optional at the level of creating an instance of the input and output spec in PyRegion.cpp. I'll update that documentation